src/tdastro/sources/bayeSN_model.py

```diff
@@ -0,0 +1,493 @@
+import h5py
+import numpy as np
+from astropy import units as u
+
+from tdastro.astro_utils.unit_utils import flam_to_fnu
+from tdastro.base_models import FunctionNode
+from tdastro.effects.extinction import ExtinctionEffect
+from tdastro.sources.physical_model import PhysicalModel
+
+
+class bayeSN_Model(PhysicalModel):
+    """A bayeSN model
+
+    The model is defined in (Mandel et al 2022) as:
+
+    flux(time, wave) = H_grid * 10 ** (-0.4 * W_grid) * 10 ** (-0.4 * (distmod _ m_abs))
+
+    This class is based on the bayesian implementation at:
+    https://github.com/bayesn/bayesn/blob/main/bayesn/bayesn_model.py
+
+    Parameterized values include:
+        * ra - The object's right ascension in degrees. [from PhysicalModel]
+        * dec - The object's declination in degrees. [from PhysicalModel]
+        * redshift - The object's redshift. [from PhysicalModel]
+        * t0 - The t0 of the zero phase, date. [from Physical Model]
+        * Amplitude - The fixed normalisation factor. [from Amplitude class]
+        * theta - The bayeSN theta parameter.
+        * Av - The bayeSN Av parameter.
+        * Rv - The bayeSN Rv parameter.
+
+    References
+    ----------
+    * BayeSN: Mandel S., 2020 - https://arxiv.org/pdf/2008.07538
+    * M20 model: Mandel et al. 2022 (MNRAS 510, 3, 3939-3966)
+    * T21 model: Thorp et al. 2021 (MNRAS 508, 3, 4310-4331) - currently unused
+    * W22 model: Ward et al. 2023 (ApJ 956, 2, 111) - currently unused
+    * Hsiao spectral template: Hsiao E. Y., 2009 - https://arxiv.org/abs/1503.02293
+
+    Attributes
+    ----------
+    _hsiao_phase:
+        The phase for hsiao template.
+    _hsiao_wave:
+        The wavelengths for hsiao template.
+    _hsiao_flux:
+        The baseline mean intrinsic SED provided by hsiao template.
+    _W0_:
+        The global W0 matrix.
+    _W1_:
+        The global W1 matrix.
+    _l_knots_:
+        The interpolation knots for wavelengths.
+    _tau_knots_:
+        The interpolation knots for phase.
+    eps: array-like
+        Set of epsilon values for each SN, describing residual colour variation.
+        Currently unused.
+
+    Parameters
+    ----------
```

Comment:
This should be a list of arguments for the __init__ function. So you should include `_M20_model_path`, `W0_filename`, etc.